### PR TITLE
fix: Add methodName and endpointName to base session.

### PR DESCRIPTION
**Pull request by Isakdl**
# Changes

Introduces `endpointName` and `methodName` on the base Session.

Related issue: https://github.com/serverpod/serverpod/issues/2450

## Pre-launch Checklist

- [ ] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [ ] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [ ] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [ ] All existing and new tests are passing.
- [ ] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

_If you have done any breaking changes, make sure to outline them here, so that they can be included in the notes for the next release._

### CODE REVIEW

File: packages/serverpod/lib/src/server/session.dart [hunk 1]
```diff
@@ -307,24 +321,21 @@ class MethodCallSession extends Session {
...
-  final String methodName;
-
-  /// The name of the endpoint that is being called.
-  final String endpointName;
-
   /// The connection id that uniquely identifies the stream.
   final UuidValue connectionId;
 
+  @override
+  String get methodName => super.methodName!;
```
Comment (SandPod) [inline packages/serverpod/lib/src/server/session.dart:328]: Suggestion: We should probably use the same funky override here as for the `MethodCallSession` to get rid of the scary `!`.

File: packages/serverpod/lib/src/server/log_manager/log_settings.dart [hunk 2]
```diff
@@ -22,71 +22,20 @@ class LogSettingsManager {
...
+    var method = session.methodName;
 
-    if (localSession is FutureCallSession) {
-      return _getLogSettingsForFutureCallSession(
-        localSession.futureCallName,
-      );
+    LogSettings? settings;
+    if (method != null) {
+      settings = _methodOverrides['$endpoint.$method'];
+      if (settings != null) return settings;
```
Comment (SandPod) [inline packages/serverpod/lib/src/server/log_manager/log_settings.dart:33]: Suggestion(NIT): Move this out of the nested if statement.

**Review by SandPod — COMMENTED** [review]
A couple of questions and minor fixes to use the same style.

File: packages/serverpod/lib/src/server/session.dart [hunk 3]
```diff
@@ -249,10 +257,16 @@ class MethodCallSession extends Session {
...
+
+  @override
+  String get endpointName => _endpointName;
 
   /// The name of the method that is being called.
-  late final String methodName;
+  late final String _methodName;
+
+  @override
+  String get methodName => _methodName;
```
[the remark this answers is not in the document]
Comment (Isakdl) [reply packages/serverpod/lib/src/server/session.dart:269]: yeah indeed! If you type cast to this subclass you are guaranteed to not get null, but the parent class can still return null (from an interface perspective).

File: packages/serverpod/lib/src/server/session.dart [hunk 4]
```diff
@@ -346,22 +357,27 @@ class StreamingSession extends Session {
...
   int currentMessageId = 0;
 
+  String _endpointName;
+
+  /// The name of the endpoint that is being called.
+  set endpointName(String endpointName) => _endpointName = endpointName;
+
+  @override
+  String get endpointName => _endpointName;
+
```
[the remark this answers is not in the document]
Comment (Isakdl) [reply packages/serverpod/lib/src/server/session.dart:371]: Yes and it is required to be overridden as the "endpointName" is now present in the base session.

**Review by SandPod — APPROVED** [review]
👍

**Pull request merged**